src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -62,7 +63,7 @@ public HubConnection(IConnection connection, IHubProtocol protocol, ILoggerFacto
             _protocol = protocol;
             _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
             _logger = _loggerFactory.CreateLogger<HubConnection>();
-            _connection.Received += OnDataReceivedAsync;
+            _connection.OnReceived((data, state) => OnDataReceivedAsync(data), null);
```

Comment:
Is `this` captured in the closure? If so you could pass `this` as the state and then cast it to `HubConnection` to call `OnDataReceivedAsync`